src/query/graphite/native/aggregation_functions.go

```diff
@@ -561,24 +561,54 @@ func applyByNode(ctx *common.Context, seriesList singlePathSpec, nodeNum int, te
 //
 //    sumSeries(foo.by-function.server1.*.cpu.load5),sumSeries(foo.by-function.server2.*.cpu.load5),...
 func groupByNode(ctx *common.Context, series singlePathSpec, node int, fname string) (ts.SeriesList, error) {
-	metaSeries := make(map[string][]*ts.Series)
-	for _, s := range series.Values {
-		parts := strings.Split(s.Name(), ".")
+	return groupByNodes(ctx, series, fname, []int{node}...)
+}
+
+func findFirstMetricExpression(seriesName string) string {
+	idxOfRightParen := strings.Index(seriesName, ")")
+	substring := seriesName[:idxOfRightParen]
+	idxOfLeftParen := strings.LastIndex(substring, "(")
+	return seriesName[idxOfLeftParen+1:idxOfRightParen]
+}
+
+func getParts(series *ts.Series) []string {
+	seriesName := series.Name()
+	if strings.Contains(seriesName, ")") {
+		seriesName = findFirstMetricExpression(seriesName)
+	}
+
+	return strings.Split(seriesName, ".")
 
-		n := node
+}
+func getAggregationKey(series *ts.Series, nodes []int) string {
+	parts := getParts(series)
+
+	var keys []string
+	for _, n := range nodes {
 		if n < 0 {
 			n = len(parts) + n
 		}
 
-		if n >= len(parts) || n < 0 {
-			return aggregate(ctx, series, fname)
+		if n < len(parts) {
+			keys = append(keys, parts[n])
```

Comment:
For graphite-web just an empty string is used if n is not within the range (see IndexError simply will "pass" which then tries to get the value from graphite tags or if not found returns empty string""):
https://github.com/graphite-project/graphite-web/blob/53a50b584016b889310eff346a2de01779613644/webapp/graphite/render/functions.py#L98-L115

To mimic same behavior instead of leaving it from the final key produced, I believe you want an else here:
```go
if n < len(parts) {
  keys = append(keys, parts[n])
} else {
  keys = append(keys, "")
}
```